authors/apps/articles/tests/test_rate_article.py

```diff
@@ -0,0 +1,159 @@
+from authors.apps.articles.tests.test_data import (
+    valid_article, valid_article_2,
+    valid_rating, slug, invalid_rating, valid_rating_2, 
+    invalid_rating2, invalid_integer, 
+    signup_and_login_user
+    )
+from authors.apps.authentication.models import User
+from authors.apps.authentication.tests.test_data import (
+    valid_user, valid_user2
+    )
+from authors.apps.authentication.tests.app_urls import (
+    register_url, login_url, activate_url, user_action_url
+    )
+
+
+class RateArticleAPIViewTestCase(TestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+
+        response = self.client.post(
+            register_url, {"user": valid_user}, format="json")
+
+        self.token = response.data["data"]["token"]
+
+        self.client.get(f"{activate_url}?token={self.token}", format='json')
+
+        response = self.client.post(
+            login_url, {"user": valid_user}, format='json')
+
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + self.token)
+
+        self.client.post('/api/articles/',
+            {"article": valid_article}, format="json")
+
+    def test_user_without_auth_cant_rate_article(self):
+        self.client.logout()
+        response = self.client.post(
+            f'/api/articles/{slug}/rate/',
+            valid_rating,
+            format="json"
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_unathenticated_user_can_see_ratings(self):
+        self.client.logout()
+        response = self.client.get(
+            f'/api/articles/{slug}/rate/',
+            valid_rating,
+            format="json"
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn(slug, response.data['slug'])
+
+    def test_athenticated_user_can_see_ratings(self):
+        response = self.client.get(
+            f'/api/articles/{slug}/rate/',
+            valid_rating,
+            format="json"
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn(slug, response.data['slug'])
+
+    def test_user_cant_rate_their_article(self):
+        response = self.client.post(
+            f'/api/articles/{slug}/rate/',
+            valid_rating,
+            format="json"
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        self.assertIn(
+            'you cant rate your own article', response.data['ratings']
+            )
+
+    def test_authenticated_user_can_rate_an_article(self):
+        self.client.logout()
+        signup_and_login_user(User, valid_user2, self.client)
+        response = self.client.post(
+            f'/api/articles/{slug}/rate/',
+            valid_rating,
+            format="json"
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_one_can_only_rate_from_0_to_5(self):
+        self.client.logout()
```

Comment:
Given that only one test requires the client logged in, you should move this log-out line and the login for the second user to set up and then have the user sign-in in that one test.